### DISCUSSION TITLE
ingesting otio timelines

### DISCUSSION THREAD

**Discussion by cgene75** [ggarra13/mrv2#320] [General]
Heya everyone,

Long time listener, first time caller.  Thank you for this excellent piece of software.

I wanted to touch base regarding the state of otio timeline ingesting.  We are trying to ingest a single single-track timeline, which i've converted to otio from both an EDL, and a fcpxml using otioconvert.

Neither managed to load into mrv1.3.0 on both linux (Rocky), or Windows 11.  Is this expected to be working currently?  have any particular export path been tested more thoroughly then others?  Any insight would be most appreciated.

Best,

G

**Comment by ggarra13** [comment]
> Neither managed to load into mrv1.3.0 on both linux (Rocky), or Windows 11. Is this expected to be working currently? have any particular export path been tested more thoroughly then others? Any insight would be most appreciated.

Yes, it is expected to work.   The .otio ingesting has been tested with the EDLs produced by mrv2 itself, those of tlRender's SampleData directory and with the ASWF .otio multitrack timeline provided by Animal Logic.

It has **NOT** been tested with actual EDLs exported from editors such as DaVinci Resolve or those produced by otioconvert.

Therefore, I suggest you attach the problematic .otio file here or, if it has sensitive info, send it to me privately at ggarra13@gmail.com.

**Comment by ggarra13** [comment]
Also, please describe the actual error you get: whether the .otio file loading fails or the actual clips used fail.

**Reply by cgene75** [reply]
Hello Gonzalo,

Thank you for looking into this issue.  I will try to be as thorough as I can.

We've exported two timelines from Flame.  One in fcpxml format, and another as a CMX3600 EDL.
On balance that's probably less critical as these elements are not ingested directly.

I then used otioconvert [OpenTimelineIO version: 0.17.0] and the OpenTimelineIO-Plugins pip packages [same version] to write two otio files named edl.otio, fcpxml respectively.

I've attempted to open both otio files on Windows11, as well as Rocky9.  Both with the same results.

when I open fcpxml.otio, I get an error message stating: 
`ERROR: [] C:\gene\fcpxml.otio: Cannot read timeline`
no images get loaded, and no clips appear.

when I open edl.otio, I get a 'timeline' that appears to be the correct length (720 frames... though truth be told I get 717, so a few frames got dropped somewhere).  If I toggle the 'edit' button, I see the first clip, but no images appear, and no subsequent shots appear on the timeline.  when I click on the first clip, mrv2 exist with:
`ERROR: Bad optional access`

I've attached both source EDLs, and otio converted data to an email I've shared with you.

Please don't hesitate to reach out if I can provide more insight or testing..

Best,

G

**Comment by cgene75** [comment]
Just an update for posterity.

It appears that there are some translation issues rooted in otioconvert (unrelated to mrv2) which are introducing a few bogus elements into the mix.  There are also some mrv2 issues that Gonzalo is addressing.

**Comment by cgene75** [comment]
Quick updated on a related topic.

Trying to load the fcpxml.otio file in windows, it appears that path mapping is ignored.

error message: 
`ERROR: [] file:///projects/project_name_redacted/caf090/images/2d/caf090_comp_v01.1001.exr: /projects/project_name_redacted/caf090/images/2d/caf090_comp_v01.1001.exr: Cannot open file: The system cannot find the path specified.`

see attached image of path mapping settings in prefs:
![path_mapping](https://github.com/user-attachments/assets/ab9da562-97c2-4dbb-ae36-7438b5fcf0db)

**Comment by ggarra13** [comment]
Yes.  This is a known limitation of path mapping.  It only works for the command-line as tlRender (Darby Johnston's library) is not aware of it.
The work-around is to use Samba.

**Comment by cgene75** [comment]
Thanks the heads up.  Samba is not currently an option for us for various storage/AD related nonsense.
The workaround is to repath the files before feeding them to mrv2.

Incidentally, I'm sure you are aware, but the frames are currently not advancing.   mrv2 only shows the first frame in the timeline.

**Reply by ggarra13** [reply]
> Incidentally, I'm sure you are aware, but the frames are currently not advancing. mrv2 only shows the first frame in the timeline.

That's another bug of otioconvert.  It is converting the sequences as OpenTimelineIO's ExternalReferences with a single frame instead of ImageSequences.

**Reply by cgene75** [reply]
Copy that.

I'll attempt to write an otio adapter that will export something that matches what mrv2 writes by default as a workaround. 
Thank you for sharing this insight.
